datumaro/plugins/icdar_format/converter.py

```diff
@@ -0,0 +1,193 @@
+            self.annotations += '\"%s\"' % ann.caption
+        self.annotations += '\n'
+
+    def write(self, path):
+        file = osp.join(path, 'gt.txt')
+        os.makedirs(osp.dirname(file), exist_ok=True)
+        with open(file, 'w') as f:
+            f.write(self.annotations)
+
+    def is_empty(self):
+        return len(self.annotations) == 0
+
+class _TextLocalizationConverter():
+    def __init__(self):
+        self.annotations = {}
+
+    def save_annotations(self, item):
+        annotation = ''
+        for ann in item.annotations:
+            if ann.type == AnnotationType.bbox:
+                annotation += ' '.join(str(p) for p in ann.points)
+                if ann.attributes and 'text' in ann.attributes:
+                    annotation += ' \"%s\"' % ann.attributes['text']
+            elif ann.type == AnnotationType.polygon:
+                annotation += ','.join(str(p) for p in ann.points)
+                if ann.attributes and 'text' in ann.attributes:
+                    annotation += ',\"%s\"' % ann.attributes['text']
+            annotation += '\n'
+        self.annotations[item.id] = annotation
+
+    def write(self, path):
+        os.makedirs(path, exist_ok=True)
+        for item in self.annotations:
+            file = osp.join(path, 'gt_' + item + '.txt')
+            with open(file, 'w') as f:
+                f.write(self.annotations[item])
+
+    def is_empty(self):
+        return len(self.annotations) == 0
+
+class _TextSegmentationConverter():
+    def __init__(self):
+        self.annotations = {}
+        self.masks = {}
+
+    def save_annotations(self, item):
+        annotation = ''
+        colormap = [(255, 255, 255)]
+        masks = [a for a in item.annotations
+            if a.type == AnnotationType.mask]
+        label_is_none = [p for p in masks
+            if p.label is None]
+        if label_is_none:
+            masks = sorted(masks, key=lambda a: a.group)
+        else:
+            masks = sorted(masks, key=lambda a: a.label)
+        group = masks[0].group
+        for ann in masks:
+            if ann.group != group or ann.group == 0:
+                annotation += '\n'
+            text = ''
+            if ann.attributes:
+                if 'text' in ann.attributes:
+                    text = ann.attributes['text']
+                if text == ' ':
+                    annotation += '#'
+                if 'color' in ann.attributes:
+                    colormap.append(ann.attributes['color'])
+                    annotation += ' '.join(str(p)
+                        for p in ann.attributes['color'])
+                else:
+                    annotation += '- - -'
+                if 'center' in ann.attributes:
+                    annotation += ' '
+                    annotation += ' '.join(str(p)
+                        for p in ann.attributes['center'])
+                else:
+                    annotation += ' - -'
+            bbox = find_mask_bbox(ann.image)
```

Comment:
Mask class has a method for getting a bbox. Also, for RLE-encoded masks it performs better.